### PR TITLE
[DEV-5140] Document GreyNoise Incoming feed EDK protocol v1

### PR DESCRIPTION
**Pull request by nathan-eiq**
Part of [DEV-5140](https://eclecticiq.atlassian.net/browse/DEV-5140)

- Added documentation to README.rst

### CODE REVIEW

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst [hunk 1]
```diff
@@ -1,39 +1,251 @@
...
-    * - API endpoint
+ ``indicator``
+
+Data Mapping
+============
+
+This extension maps data from GNQL JSON to EIQ JSON format.
+
+
+.. list-tables::
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst:81]: add one more space!

`.. list-tables::` -> `..  list-tables::`

need to align indentation for tables to properly render :sweat_smile:

**Comment by zed-eiq** [comment]
other than my comment about indentation, it looks good! thank you! :pray:

**Comment by zed-eiq** [comment]
also please follow the PR naming convention `[<ticket#>] <purpose of PR>`!

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst (hunk 1 above)
Comment (zed-eiq) [reply extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst:81]: just realised -- like in #118 , this should be `list-table` :grin:

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst [hunk 2]
```diff
@@ -1,39 +1,251 @@
...
+    :align: left
+    :header-rows: 1
+    :stub-columns: 1
+
+    "title","name","type","required","description","default"
+    "API URL","api_url","string","true","The URL pointing to the API endpoint exposing the service that makes the data available for retrieval through the feed.","https://api.greynoise.io"
+    "API Key","api_key","password","true","API Key","***"
+    "SSL verification","custom_ssl","boolean","true","SSL verification","true"
+    "Path to SSL certificate file","verify","file","false","Path to SSL certificate file",""
+    "Classification","classification","array","true","Include indicators with selected classification",""
```
Comment (petarc-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst:60]: default="malicious"

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst [hunk 3]
```diff
@@ -1,39 +1,251 @@
...
+
+The following classification can be selected:
+
+- ``Malicious``
+- ``Benign``
+- ``Unknown``
+
+Supported Observable types
```
Comment (petarc-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst:68]: This section is applicable only to Enrichers.

**Review by petarc-eiq — CHANGES_REQUESTED** [review]

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst (hunk 3 above)
Comment (nathan-eiq) [reply extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst:68]: I see what you mean. That is indeed where the section name comes from. The idea behind this section is to tell the users what entities they will get from the extension. I'll change that to make clearer

Comment (zed-eiq) [reply extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst]: just remove! see https://github.com/eclecticiq/extensions-store/pull/120/files/c725ca60ad05161cfd7e3173a767b6ce4b96f33b..a95cf3fa7ad3809f7bb8deaa1e7c69b28afb2e59#r1206994028

Comment (nathan-eiq) [reply extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst]: Done

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst [hunk 4]
```diff
@@ -1,39 +1,244 @@
...
+    :header-rows: 1
+    :stub-columns: 1
+
+    * - Field Name
+      - GNQL JSON
+      - EIQ JSON
+
+    * - country
+      - ``data.metadata.country``
+      - ``<country>``
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst:202]: these EIQ JSON values should be JSONpath-like things, right? instead, i think you've taken the 'value' portion of the mapping where we need the JSONpath/key of the EIQ JSON that this data is mapped to.

Same for all other EIQ JSON table cells here and probably in the other GreyNoise exts

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst [hunk 5]
```diff
@@ -1,39 +1,244 @@
...
+      - ``data.metadata.tor``
+      - ``<boolean value>``
+
+    * - Category
+      - ``data.metadata.category``
+      - ``<category>``
+
+  
+Extracts table
+--------------
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst:190]: rename to "Map to observables"

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst [hunk 6]
```diff
@@ -1,39 +1,244 @@
...
+Tags table
+----------
+
+..  list-table::
+    :header-rows: 1
+    :stub-columns: 1
+
+    * - Field Name
+      - GNQL JSON
+      - Data format
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst:146]: remove the data format column entirely

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst [hunk 7]
```diff
@@ -1,39 +1,244 @@
...
-Configuration options
+    * - Low Impact
+      - ``benign``
+      - ``Low``
+
+    * - High Impact
+      - ``malicious``
+      - ``High``
+
+Tags table
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst:137]: Rename "Map to tags"

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst [hunk 8]
```diff
@@ -1,39 +1,244 @@
...
+----------
+
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst:139]: Unclear that this section is describing data that is essentially just ingested as tags.

This means that
- when the feed is run, the objects generated (`indicator` objects in this case) are the anchoring object, and all the data here are added as tags on the anchoring object
- that "data formats" here are irrelevant/incorrect.

you can also see in the MR here that for specific GNQL JSON fields ingested as tags, we format these tags in a specific way: https://github.com/eclecticiq/extensions-store/pull/114/files#diff-7940a9fcff55a64b8bd438807e6103e158f8736c01b7b64f5a2957003bf35bceR116

e.g. for `data.spoofable`, we add a tag that looks like `Spoofable: <value>`

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst [hunk 9]
```diff
@@ -1,39 +1,244 @@
...
+The following classification can be selected:
+
+- ``Malicious``
+- ``Benign``
+- ``Unknown``
+
+Data Mapping
+============
+
+This extension maps data from GNQL JSON to EIQ JSON format.
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst:71]: This extension processes responses from GNQL queries to produce EclecticIQ Indicator entities.
The following sections show how GNQL JSON is mapped to Indicator entities in EIQ JSON.

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst [hunk 10]
```diff
@@ -1,39 +1,244 @@
...
 ..  list-table::
     :header-rows: 1
-    :align: left
     :stub-columns: 1
 
-    * -
-      - Specifications
+    * - Field Name
+      - GNQL JSON
+      - EIQ JSON
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst:80]: - your first column is actually the EIQ JSON field name
- your last column is a description of the data

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+      - See `Map to Observables`_ 
+
+Impact table
+------------
+
+..  list-table::
+    :header-rows: 1
+    :stub-columns: 1
+
+    * - Description
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst:120]: Remove the `Description` column

- Doesn't add value here
- `Description` should never be the first column. You want a short, maybe sortable value for the first column of the table (i.e. "primary key-ish" value)

File: extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+============
+
+This extension maps data from GNQL JSON to EIQ JSON format.
+
+
+..  list-table::
+    :header-rows: 1
+    :stub-columns: 1
+
+    * - Field Name
```
Comment (zed-eiq) [inline extensions/greynoise/incoming_feeds/greynoise_noise_incoming_feed_v1/README.rst:77]: can't remember where the feedback was given elsewhere, but basically the table headings here don't make sense.

- first column is EIQ JSON fields
- third column can be named `Description`

**Review by zed-eiq — CHANGES_REQUESTED** [review]
More changes!
Also, please rebase your branch on petar's latest changes :pray:

**Comment by zed-eiq** [comment]
thanks! the README.rst file looks good now!

can you fix the rebase?
- you might have rebased it on an outdated (local) version of the target branch
- or you might have rebased on the wrong branch

alternatively, close this MR and start a new one referencing this one :+1:

**Pull request closed without merging**